target/bin/delmailuser

```diff
@@ -90,73 +90,72 @@ fi
 (
   flock -e 200
   ERROR=false
+  USER="${*}"
 
-  for USER in "${@}"
-  do
-    # very simple plausibility check
-    [[ ${USER} != *'@'*'.'* ]] && errex "No valid address: ${USER}"
+  # very simple plausibility check
+  [[ ${USER} != *@*.* ]] && errex "No valid address: ${USER}"
 
-    declare -a MAILARR
-    MAILARR[0]="${USER%@*}"
-    MAILARR[1]="${USER#*@}"
+  unset MAILARR
```

Comment:
Why is that needed? MAILARR is not set before?

Reply:
In my tests in #1808 the array was (for some reason) already filled.

If I remember correctly, the unset of some variables in the current version (in this PR) should not interfere with something else since it's locked behind the subshell access (?). I'm deliberately unsetting `EMAIL USER DOMAIN` even though it might strictly speaking not be necessary.

Reply:
if `MAILARR` is already filled, but not expected to be, we should find out why.

Only this two files are also involved.
```
setup.sh
helper-functions.sh
```
In none of them I could find a reference for `MAILARR`. Can you double check, that `MAILARR` is really set?

Reply:
As you mentioned, I got rid of it altogether. I think we can resolve this?

Reply:
Why this:  `unset EMAIL USER DOMAIN` ?

Reply:
Probably not actually needed. Shall I remove it?

Reply:
Yes

Reply:
I guess, add/delete an email address is covered by bats tests? Otherwise, a quick test setup to see if add/delete works as intended wouldn't hurt.